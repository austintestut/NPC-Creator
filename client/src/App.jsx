import React from 'react';
import styled from 'styled-components';
import axios from 'axios';
import AddNewNPCButton from './AddNewNPCButton';
import AddNewNPCForm from './AddNewNPCForm';
import NPCCardContainer from './NPCCardContainer';
import EditNPCForm from './EditNPCForm';
import helpers from './helperData';


class App extends React.Component {
  constructor() {
    super();
    this.state = {
      npcData: [],
      npcFormName: '',
      npcFormRace: '',
      npcFormDemeanor: '',
      npcFormQuality: '',
      npcFormNotes: '',
      addFormShowing: false,
      editFormShowing: false,
      editID: null
    }
    this.getAllNPCs = this.getAllNPCs.bind(this);
    this.generateNPC = this.generateNPC.bind(this);
    this.updateNameForm = this.updateNameForm.bind(this);
    this.updateRaceForm = this.updateRaceForm.bind(this);
    this.updateDemeanorForm = this.updateDemeanorForm.bind(this);
    this.addNPC = this.addNPC.bind(this);
    this.showAddForm = this.showAddForm.bind(this);
    this.showEditForm = this.showEditForm.bind(this);
    this.cancelEdit = this.cancelEdit.bind(this);
    this.cancelAdd = this.cancelAdd.bind(this);
    this.updateNPC = this.updateNPC.bind(this);
    this.deleteNPC = this.deleteNPC.bind(this);
    this.updateNotesForm = this.updateNotesForm.bind(this);
    this.updateQualityForm = this.updateQualityForm.bind(this);
  }

  componentDidMount() {
    this.getAllNPCs();
  }

  getAllNPCs() {
    axios.get('/npcs')
      .then((data) => {
        this.setState({
          npcData: data.data
        });
      })
      .catch((err) => {
        console.log('err getting NPCs --client');
      })
  }

  addNPC() {
    if (
      this.state.npcFormName === '' ||
      this.state.npcFormRace === '' ||
      this.state.npcFormDemeanor === ''
    ) {
      window.alert('Cannot submit blank NPC!\nThat defeats the purpose of this app!\n(Quality can be blank)');
      return;
    }
    axios.post('/npcs', {
      name: this.state.npcFormName,
      race: this.state.npcFormRace,
      demeanor: this.state.npcFormDemeanor,
      quality: this.state.npcFormQuality
    })
      .then((res) => {
        document.getElementById('nameInput').value = '';
        document.getElementById('raceInput').value = '';
        document.getElementById('demeanorInput').value = '';
        document.getElementById('qualityInput').value = '';
        this.setState({
          npcFormName: '',
          npcFormRace: '',
          npcFormDemeanor: ''
        });
        this.getAllNPCs();
      })
      .catch((err) => {
        console.log('err saving NPC --client');
      })
  }

  generateNPC() {
    let randomRace = helpers.races[this.randomNumberGenerator(0, 8)];
    let raceAPIParam = helpers.raceAPIStyle[randomRace];
    let randomDemeanor = helpers.demeanors[this.randomNumberGenerator(0, 39)];
    let randomQuality = helpers.qualities[this.randomNumberGenerator(0, 37)];
    axios.get(`/name/${raceAPIParam}`)
      .then((name) => {
        document.getElementById('nameInput').value = name.data;
        document.getElementById('raceInput').value = randomRace;
        document.getElementById('demeanorInput').value = randomDemeanor;
        document.getElementById('qualityInput').value = randomQuality;
        this.setState({
          npcFormName: name.data,
          npcFormRace: randomRace,
          npcFormDemeanor: randomDemeanor,
          npcFormQuality: randomQuality
        })
      })
      .catch((err) => {
        console.log('err generating NPC');
      })
  }

  randomNumberGenerator(min, max) {
    // inclusive of min and max
    min = Math.ceil(min);
    max = Math.floor(max);
    return Math.floor(Math.random() * (max - min + 1) + min);
  }

  updateNPC() {
    if (
      this.state.npcFormName === '' ||
      this.state.npcFormRace === '' ||
      this.state.npcFormDemeanor === ''
    ) {
      window.alert('Cannot submit blank NPC!\nThat defeats the purpose of this app!');
      return;
    }
    axios.put('/npcs', {
      id: this.state.editID,
      name: this.state.npcFormName,
      race: this.state.npcFormRace,
      demeanor: this.state.npcFormDemeanor,
      notes: this.state.npcFormNotes,
      quality: this.state.npcFormQuality
    })
    .then((response) => {
      this.setState({
        editFormShowing: false,
        npcFormName: '',
        npcFormRace: '',
        npcFormDemeanor: '',
        npcFormNotes: '',
        npcFormQuality: ''
      });
      this.getAllNPCs();
    })
  }

  deleteNPC() {
    axios.put('/npcs/delete', {
      id: this.state.editID
    })
    .then((response) => {
      this.setState({
        editFormShowing: false,
        npcFormName: '',
        npcFormRace: '',
        npcFormDemeanor: ''
      });
      this.getAllNPCs();
    })
  }

  updateNameForm(e) {
    this.setState({
      npcFormName: e.target.value
    });
  }

  updateRaceForm(e) {
    this.setState({
      npcFormRace: e.target.value
    });
  }

  updateDemeanorForm(e) {
    this.setState({
      npcFormDemeanor: e.target.value
    });
  }

  updateNotesForm(e) {
    this.setState({
      npcFormNotes: e.target.value
    });
  }

  updateQualityForm(e) {
    this.setState({
      npcFormQuality: e.target.value
    });
  }

  showAddForm() {
    this.setState({
      addFormShowing: true
    });
  }

  showEditForm(id, name, race, demeanor, notes, quality) {
    this.setState({
      editFormShowing: true,
      editID: id,
      npcFormName: name,
      npcFormRace: race,
      npcFormDemeanor: demeanor,
      npcFormNotes: notes,
      npcFormQuality: quality
    }, () => {
      document.getElementById('editNameInput').value = name;
      document.getElementById('editRaceInput').value = race;
      document.getElementById('editDemeanorInput').value = demeanor;
      document.getElementById('editNotesInput').value = notes;
      document.getElementById('editQualityInput').value = quality;
    })
  }

  cancelEdit() {
    this.setState({
      editFormShowing: false,
      editID: null,
      npcFormName: '',
      npcFormRace: '',
      npcFormDemeanor: ''
    })
  }

  cancelAdd() {
    this.setState({
      addFormShowing: false,
      npcFormName: '',
      npcFormRace: '',
      npcFormDemeanor: ''
    })
  }

  render() {
    return (
      <>
        <h2>NPC Creator</h2>
        <h4><i>Stop naming your NPCs Bob!</i></h4>
        <AddNewNPCButton
          showAddForm={this.showAddForm}
        />
        {this.state.addFormShowing && <AddNewNPCForm
          generateNPC={this.generateNPC}
          updateNameForm={this.updateNameForm}
          updateRaceForm={this.updateRaceForm}
          updateDemeanorForm={this.updateDemeanorForm}
          updateQualityForm={this.updateQualityForm}
          addNPC={this.addNPC}
          cancelAdd={this.cancelAdd}
        />}
        <h2>My NPCs</h2>
        <NPCCardContainer
          npcData={this.state.npcData}
          showEditForm={this.showEditForm}
        />
        {this.state.editFormShowing && <EditNPCForm
          updateNameForm={this.updateNameForm}
          updateRaceForm={this.updateRaceForm}
          updateDemeanorForm={this.updateDemeanorForm}
          cancelEdit={this.cancelEdit}
          updateNPC={this.updateNPC}
          deleteNPC={this.deleteNPC}
          updateNotesForm={this.updateNotesForm}
        />}
      </>
    )
  }
}

export default App;