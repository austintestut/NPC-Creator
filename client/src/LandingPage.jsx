import React from "react";
import Footer from "./Footer.jsx";
import Title from "../../public/images/npcc.png";

const LandingPage = ({ authenticateUser, makeSessionless }) => (
  <div data-testid="landing-page" id="landing">
    <img id="landing-title-pic" src={Title} />
    <span id="landing-description">
      Create and save new Non-Player Characters on the fly!
    </span>
    <div className="login-button-container">
        <a href="/google" className="login-anchor">
          <button type="button" id="login-button">
            <div className="google-icon-wrapper">
              <img
                className="google-icon"
                src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
              />
            </div>
            <span className="login-text">Log in with Google</span>
          </button>
        </a>
      <span id="login-separator">- or -</span>
      <button id="sessionless-login" onClick={makeSessionless}>
        Continue without an account
      </button>
    </div>
    <Footer />
  </div>
);

export default LandingPage;
