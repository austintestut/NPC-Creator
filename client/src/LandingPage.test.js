import App from "./App.jsx";
import LandingPage from "./LandingPage.jsx";
import React from "react";
import ReactDOM from "react-dom";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";

import renderer from "react-test-renderer";

afterEach(cleanup);

test("Landing Page renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<LandingPage></LandingPage>, div);
});

test("Landing Page renders correctly when user is not authenticated", () => {
  const { getByTestId } = render(
    <LandingPage authenticateUser={false} />
  );
  // expect(getByTestId("landing-page"));
});

test("Landing Page matches snapshot", () => {
  const tree = renderer
    .create(<LandingPage authenticateUser={false} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
