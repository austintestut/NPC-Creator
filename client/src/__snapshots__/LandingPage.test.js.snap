// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page matches snapshot 1`] = `
<div
  data-testid="landing-page"
  id="landing"
>
  <img
    id="landing-title-pic"
    src="test-file-stub"
  />
  <span
    id="landing-description"
  >
    Create and save new Non-Player Characters on the fly!
  </span>
  <div
    className="login-button-container"
  >
    <a
      className="login-anchor"
      href="/google"
    >
      <button
        id="login-button"
        type="button"
      >
        <div
          className="google-icon-wrapper"
        >
          <img
            className="google-icon"
            src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
          />
        </div>
        <span
          className="login-text"
        >
          Log in with Google
        </span>
      </button>
    </a>
    <span
      id="login-separator"
    >
      - or -
    </span>
    <button
      id="sessionless-login"
    >
      Continue without an account
    </button>
  </div>
  <footer>
    <span>
      This application is being constantly updated and is in no way in its final state. This application uses cookies for authenticating user sessions. By using this application you agree to use of cookies in this fashion.
    </span>
  </footer>
</div>
`;
